Missing <cassert> include?

<!-- This is a channel to report bugs/issues, not a support channel to help install/use/debug your own code. We'd love to help, but just don't have the bandwidth. Please post questions in the GTSAM Google group (https://groups.google.com/forum/#!forum/gtsam-users) -->

<!--Please only submit issues/bug reports that come with enough information to reproduce them, ideally a unit test that fails, and possible ideas on what might be wrong. -->

<!-- Even better yet, fix the bug and/or documentation, add a unit test, and create a pull request! -->

## Description

I am experiencing a compile error when building gtsam from CMake, but not when doing it manually. Thus I am not sure when this gets triggered, but either way it could be a missing include.

## Steps to reproduce

1.  Create a CMakeLists file like this:

```
project(PointCloudIntegrator LANGUAGES CXX)

include(ExternalProject)

# Set the URL and desired GTSAM version/tag
set(GTSAM_REPO "https://github.com/borglab/gtsam.git")
set(GTSAM_VERSION "4.2a9")

# Set the installation directory for GTSAM
set(GTSAM_INSTALL_DIR "${CMAKE_BINARY_DIR}/gtsam_install")

# Use ExternalProject to download and build GTSAM
ExternalProject_Add(
    gtsam
    PREFIX ${CMAKE_BINARY_DIR}/gtsam
    GIT_REPOSITORY ${GTSAM_REPO}
    GIT_TAG ${GTSAM_VERSION}
    GIT_PROGRESS ON
    CMAKE_ARGS
        -DCMAKE_INSTALL_PREFIX=${GTSAM_INSTALL_DIR}
        -DGTSAM_BUILD_EXAMPLES_ALWAYS=OFF
        -DGTSAM_BUILD_TESTS=OFF
        -DGTSAM_WITH_TBB=OFF
        -DGTSAM_BUILD_WITH_MARCH_NATIVE=ON
        -DGTSAM_USE_SYSTEM_EIGEN=ON
        -DCMAKE_CXX_STANDARD=17
        -DCMAKE_VERBOSE_BUILD=ON
        -DCMAKE_VERBOSE_MAKEFILE=ON
)
```
2. `mkdir build && cd build`
3. `cmake .. && make`

This results in

```
<project>/build/gtsam/src/gtsam/gtsam/base/Vector.cpp:20:
<project>build/gtsam/src/gtsam/gtsam/base/Vector.h:224:3: error: use of undeclared identifier 'assert'; did you mean 'mpl_::assert_'?
  224 |   assert (y.size()==x.size());
      |   ^
/opt/homebrew/include/boost/mpl/assert.hpp:97:8: note: 'mpl_::assert_' declared here
```

Meanwhile the same cmake args to a plain manual build do not cause this issue.

## Expected behavior

No compile error.

## Environment

macOS 15.1.1, cmake version 3.31.0, apple clang 16.0.0.

--- a/gtsam/base/Vector.h
+++ b/gtsam/base/Vector.h
@@ -16,6 +16,7 @@
  * @author  Frank Dellaert
  * @author  Alex Hagiopol
  * @author  Varun Agrawal
+ * @author  Fan Jiang
  */
 
 // \callgraph
@@ -193,14 +194,12 @@ GTSAM_EXPORT Vector ediv_(const Vector &a, const Vector &b);
  */
 template<class V1, class V2>
 inline double dot(const V1 &a, const V2& b) {
-  assert (b.size()==a.size());
   return a.dot(b);
 }
 
 /** compatibility version for ublas' inner_prod() */
 template<class V1, class V2>
 inline double inner_prod(const V1 &a, const V2& b) {
-  assert (b.size()==a.size());
   return a.dot(b);
 }
 

